docs/docs/05-migrations/13-v8-to-v9.md

```diff
@@ -24,7 +24,9 @@ There are four sections based on the four potential user groups of this document
 
 ## IBC Apps
 
-- No relevant changes were made in this release.
+### API removals
+
+The `exported.ChannelI` interface has been removed. Please use the concrete type. 
```

Comment:
My approach is to deprecate changes which are disruptive to users. Users shouldn't be using the `exported.ChannelI` interface (it was never intended to be used externally), as they likely need to cast to `channeltypes.Channel`, but if they are, the only diffs they have is changing from `exported.ChannelI` to `channeltypes.Channel`

I agree a deprecation policy would be nice, but I feel it might be good to give ourselves some flexibility in deciding on a case by case basis, as some removals are more disruptive than others

Reply:
Some other points that come to mind:
* these changes can be applied pre-v9 (users can make the changes in a pr before bumping to v9, so the depreciation notice could actually be posted in the v8 release, but honestly don't think anyone will notice until the compiler breaks)
* no functional changes

I feel those are 2 good points for justifying immediate removal. Maybe we can start bundling justifications in a document so it can be more formalized